linode/manager
# test: [M3-8013] - Upgrade to Cypress v13.11.0

## Description 📝
This upgrades Cypress and related dependencies to their latest versions. Check out [Cypress's changelog](https://docs.cypress.io/guides/references/changelog) for a more detailed overview of the changes since 13.5, but I've noted some of the most noteworthy changes:

- Improve type definitions for `cy.wait()` and `cy.intercept()` ([see entry](https://docs.cypress.io/guides/references/changelog#:~:text=Enhanced%20the%20type%20definitions%20available%20to%20cy.intercept()%20and%20cy.wait().%20The%20body%20property%20of%20both%20the%20request%20and%20response%20in%20an%20interception%20can%20optionally%20be%20specified%20with%20user%2Ddefined%20types.%20Addresses%20%2329507.))
- Support Vite 5 (this is especially relevant for #10134) ([see entry](https://docs.cypress.io/guides/references/changelog#:~:text=Added%20support%20for%20vite%20v5%20to%20%40cypress/vite%2Ddev%2Dserver.%20Addresses%20%2328347.))
- Fixes a regression introduced in 13.6 that would cause Cypress to crash mid-run; this had been previously blocking our upgrade ([see entry](https://docs.cypress.io/guides/references/changelog#:~:text=Fixed%20a%20regression%20introduced%20in%2013.6.0%20where%20Cypress%20would%20occasionally%20exit%20with%20status%20code%201%2C%20even%20when%20a%20test%20run%20was%20successful%2C%20due%20to%20an%20unhandled%20WebSocket%20exception%20(Error%3A%20WebSocket%20connection%20closed).%20Addresses%20%2328523.))
- Fixes a bug with `cy.wait` that sometimes causes an error to be thrown. We've been getting unexplained test failures that are similar to the error described in the [related PR](https://github.com/cypress-io/cypress/issues/28233), so this upgrade may fix them ([see entry](https://docs.cypress.io/guides/references/changelog#:~:text=Fixed%20an%20issue%20where%20cy.wait()%20would%20sometimes%20throw%20an%20error%20reading%20a%20property%20of%20undefined%20when%20returning%20responses.%20Fixes%20%2328233.))

## Changes  🔄
List any change relevant to the reviewer.
- Upgrades `cypress` to 13.11, and updates our `Dockerfile` accordingly
- Upgrades `cypress-axe` to `1.5.x`, `cypress-real-events` to `1.12.x`, and installs the latest patch versions for `@testing-library/cypress` and `cypress-file-upload`

## How to test 🧪
We can rely on CI for this. Because we're skipping several versions, I plan to run the pipeline several times to ensure there aren't any changes that might impact our test stability.
